.mockery.yaml

```diff
@@ -0,0 +1,36 @@
+with-expecter: false
+disable-version-string: true
+outpkg: "mocks"
+dir: "{{.InterfaceDir}}/mocks/"
+filename: "{{ .InterfaceName | snakecase }}.go"
+packages:
+  github.com/jaegertracing/jaeger/pkg/es:
+    config:
+      all: true
+      mockname: "{{.InterfaceName}}"
+      filename: "{{.InterfaceName}}.go"
+  github.com/jaegertracing/jaeger/pkg/es/client:
+    interfaces:
+      IndexAPI:
+        config:
+          filename: "index_client.go"
```

Comment:
there seems to be a lot more details (or customizations) here compared to what we had in the makefile, are they all required? Can we rely more on the defaults?

also, as part of the PR we need to change the target in the makefile, so that CI can validate these changes

Reply:
sure! I can clean up all except the details for `github.com/jaegertracing/jaeger/pkg/es/client` because these files don't match the rest (eg. mocks start with a "Mock" prefix).

Reply:
about CI, are there any further tests you recommend running locally aside from `make test-ci`?

Reply:
why is `es/client` special? I prefer everything to be uniform, if this package diverges in conventions from the others, we should just fix that.

Reply:
I have no idea; it seems to follow mockery conventions, though. I will update the files